lib/elasticsearch_models/base.rb

```diff
@@ -36,6 +37,17 @@ def build!(*params)
         model
       end
 
+      def save_model!(model)
+        model._id.nil? or raise "Model already saved, cannot be saved again"
+        model.rehydration_class = type
+        model.query_types       = query_types
+        model.validate!
```

Comment:
This logic is duplicated in `build!`.  We could consider something like `assign_internal_fields` (so at least it's similar to `assign_metadata_fields`)?

```
model.rehydration_class = type
model.query_types       = query_types
```

Reply:
I was able to move this to the constructor. (It was more of a pain than it should have been.)